### ISSUE TITLE
More specified dependency information

### ISSUE THREAD

**Issue by LYC-vio** [micahvista/MAMnet#2]
Hi, I am trying to test your tool in my environment, but have encountered several problems regarding the package versions.
I first tried to run MAMnet with python3.8, numpy1.21, tensorflow2.4.1, but get the following error:
```
Traceback (most recent call last):
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/data/maiziezhou_lab/Yichen/Softwares/MAMnet/MAMnet.py", line 431, in call_sv
    model = main_model(mc)
  File "/data/maiziezhou_lab/Yichen/Softwares/MAMnet/MAMnet.py", line 142, in main_model
    encoded_frames = layers.Bidirectional(tf.keras.layers.LSTM(256, return_sequences= True))(encoded_frames)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/wrappers.py", line 539, in __call__
    return super(Bidirectional, self).__call__(inputs, **kwargs)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 951, in __call__
    return self._functional_construction_call(inputs, args, kwargs,
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 1090, in _functional_construction_call
    outputs = self._keras_tensor_symbolic_call(
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 822, in _keras_tensor_symbolic_call
    return self._infer_output_signature(inputs, args, kwargs, input_masks)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 863, in _infer_output_signature
    outputs = call_fn(inputs, *args, **kwargs)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/wrappers.py", line 652, in call
    y = self.forward_layer(forward_inputs,
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/recurrent.py", line 660, in __call__
    return super(RNN, self).__call__(inputs, **kwargs)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 1012, in __call__
    outputs = call_fn(inputs, *args, **kwargs)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/recurrent_v2.py", line 1157, in call
    inputs, initial_state, _ = self._process_inputs(inputs, initial_state, None)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/recurrent.py", line 859, in _process_inputs
    initial_state = self.get_initial_state(inputs)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/recurrent.py", line 642, in get_initial_state
    init_state = get_initial_state_fn(
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/recurrent.py", line 2506, in get_initial_state
    return list(_generate_zero_filled_state_for_cell(
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/recurrent.py", line 2987, in _generate_zero_filled_state_for_cell
    return _generate_zero_filled_state(batch_size, cell.state_size, dtype)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/recurrent.py", line 3003, in _generate_zero_filled_state
    return nest.map_structure(create_zeros, state_size)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/util/nest.py", line 659, in map_structure
    structure[0], [func(*x) for x in entries],
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/util/nest.py", line 659, in <listcomp>
    structure[0], [func(*x) for x in entries],
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/keras/layers/recurrent.py", line 3000, in create_zeros
    return array_ops.zeros(init_state_size, dtype=dtype)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/util/dispatch.py", line 201, in wrapper
    return target(*args, **kwargs)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/ops/array_ops.py", line 2819, in wrapped
    tensor = fun(*args, **kwargs)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/ops/array_ops.py", line 2868, in zeros
    output = _constant_if_small(zero, shape, dtype, name)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/ops/array_ops.py", line 2804, in _constant_if_small
    if np.prod(shape) < 1000:
  File "<__array_function__ internals>", line 5, in prod
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 3051, in prod
    return _wrapreduction(a, np.multiply, 'prod', axis, dtype, out,
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 86, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
  File "/home/liuy120/.conda/envs/MAMnet/lib/python3.8/site-packages/tensorflow/python/framework/ops.py", line 852, in __array__
    raise NotImplementedError(
NotImplementedError: Cannot convert a symbolic Tensor (bidirectional/forward_lstm/strided_slice:0) to a numpy array. This error may indicate that you're trying to pass a Tensor to a NumPy call, which is not supported
```

some answers said that this could be solved by downgrading numpy to 1.19, however, when running with numpy 1.19, another error information popped out:
```
Traceback (most recent call last):
  File "/data/maiziezhou_lab/Yichen/Softwares/MAMnet/MAMnet.py", line 240, in <module>
    import mamnet
ImportError: numpy.core.multiarray failed to import
``` 

I've followed the dependencies instruction provided in your readme, but it seems that the version information is not sufficient for users to easily install MAMnet. Would you please make it more specific?

Best,
Yichen

**Comment by micahvista** [comment]
The 'requirements.txt' has been added to this repository, you can easily install it with command 'pip install -r requirements.txt'.

**Comment by LYC-vio** [comment]
Hi, after searching for the reason of the "Cannot convert a symbolic Tensor" issue, I found that this could be solved by downgrading numpy to 1.19.5 or use tensorflow 2.6+ (https://stackoverflow.com/a/65993776)
